### DISCUSSION TITLE
Available Macro and Tap Dance Entries

### DISCUSSION THREAD

**Discussion by xfo-0** [vial-kb/vial-gui#93] [Q&A]
Is there a way to increase the amount of Macro/Tap Dance entries for use in vial-gui? From what I've gathered, macros are stored to the available EEPROM which is usually limited and by default is 16. qmk's docs declare that when defined by a keymap.json, it's possible to store up to 32 and basically more until capacity is reached. By flashing empty definitions, would it increase this amount for vial-gui? After attempting to do so, it seems that this is set to 16 and is maybe handled differently by qmk than only the keymap.json? Also, given that macro definitions in vial-gui are not actually written until saved, and that memory usage is already read and provided on screen indicating whether or not it will be over memory, would being able to add or remove slots based on this free space be a possibility within vial-gui in the future? It seem that since there's already some prevention for going over in place, this could potentially allow for much more flexibility.

**Comment by xyzz** [comment]
Currently you cannot increase the number of macros, this is something hardcoded in VIA code; if this is something you need I suggest filing a feature request at https://feedback.vial.today/

To increase the number of tap dance you can add `#define VIAL_TAP_DANCE_ENTRIES XXX` to your `config.h` file - other defines documented here: https://get.vial.today/docs/firmware-size.html

Dynamic memory allocation for things like tap dance/combos/etc would require major design changes and is not currently something planned.
